### PR TITLE
Make a pass over setup, docs, and top-level things

### PR DESCRIPTION
**Pull request by rocky**
* README.rst: move customization from main.py here
* Makefile: LoadModule name has changes
* __init__.py: list imports, go over docstring which forms chapter information
* main.py: numerous small doc changes; fix some type errors.
* setup.py and spacy, langid, and pyenchant are no longer optional. Note pyenchant rather than enchant is now used.

### CODE REVIEW

**Comment by rocky** [comment]
As soon as this clears I will be merging to try to get in the next dockerhub image. As before comments and corrections will be handled in future PRs.

File: pymathics/natlang/main.py [hunk 1]
```diff
@@ -484,12 +451,12 @@ def eval(self, text: String, word, evaluation: Evaluation, options: dict):
...
-            text = token.text
+            token_text = token.text
             if ignore_case:
-                text = text.lower()
-            if text in words:
+                token_text = token_text.lower()
+            if token_text in words:
                 n += 1
-        return Expression(SymbolN, Integer(n) / Integer(len(doc)))
+        return eval_N(Integer(n) / Integer(len(doc)), evaluation)
```
Comment (mmatera) [inline pymathics/natlang/main.py:459]: Here we could use `eval_N`

Comment (rocky) [reply pymathics/natlang/main.py:459]: I don't understand - that is what is currently used.

Comment (mmatera) [reply pymathics/natlang/main.py:459]: Sorry, I didn't see the change.

**Pull request merged**